### ISSUE TITLE
[App] custom charts (eg. Confusion Matrix and ROC) show "no data available"

### ISSUE THREAD

**Issue by daia99** [wandb/client#2051]
**Describe the bug**
As of 2 days ago before making this issue, the custom charts I used no longer display data using wandb.log() in the web client.

**To Reproduce**

The following calls are used from following the docs:
-  `wandb.log({"roc" : wandb.plot.roc_curve(ground_truth,
           predictions, labels=["a", "b"])})`
- `wandb.log({"conf_mat" : wandb.plot.confusion_matrix(
                        probs=None,
                        y_true=ground_truth,
                        preds=predictions,
                        class_names=["a", "b"]})`

Using wandb with PyTorch Lightning, but with direct calls from wandb package.

Tables under the 'Media' tab are created with values for each custom chart, but not recognized as data to be displayed in 'Custom Charts'. Instead, custom charts show "no data available".

This also occurred after logging to a different wandb project ID.

**Expected behavior**
Values to be plotted in the respective custom chart formats, updated after every custom chart log call for the same run.

![wandb_custom_charts](https://user-images.githubusercontent.com/33094749/114014041-187c1700-9860-11eb-892b-994db2e6b7a4.png)

**Comment by ariG23498** [comment]
Hey @daia99 
Would it be possible for you to link us to a colab notebook for reproduction?

**Comment by daia99** [comment]
Hi @ariG23498 

I looked up these notebook examples from [here](https://github.com/wandb/examples/tree/master/colabs/wandb-log):

- [Confusion Matrix](https://colab.research.google.com/github/wandb/examples/blob/master/colabs/wandb-log/Log_a_Confusion_Matrix_with_W%26B.ipynb#scrollTo=7097i45lx8sf)
- [ROC](https://colab.research.google.com/github/wandb/examples/blob/master/colabs/wandb-log/Plot_ROC_Curves_with_W%26B.ipynb)

Let me know if that helps

**Comment by ariG23498** [comment]
Hey @daia99 
I did not find any error while running the linked examples
The custom charts are well visible in the UI.
Would you like to add anything to the process that could reproduce the behaviour that you are presently seeing?

**Comment by IgorHoholko** [comment]
@ariG23498 Absolutely the same error.
![image](https://user-images.githubusercontent.com/27057946/115187526-234e6b80-a0ec-11eb-9068-4064c873d70c.png)
![image](https://user-images.githubusercontent.com/27057946/115187551-2ea19700-a0ec-11eb-8af4-7c0b689bb92e.png)

**Comment by IgorHoholko** [comment]
When I'm trying to do this, the log takes a lot of time and produce these errors
![image](https://user-images.githubusercontent.com/27057946/115187808-a243a400-a0ec-11eb-82b2-2d1e74ac86fc.png)

**Comment by vanpelt** [comment]
@daia99 can you share a code snippet or colab that causes this error for you?

**Comment by vanpelt** [comment]
@IgorHoholko in your screenshot you're specifying "rc_table" as the key, but it looks like you're logging the data under "rrlocc".  Also, are you able to share a colab or simple reproduction case?

**Comment by IgorHoholko** [comment]
@vanpelt I logged a lot of examples :) That is okay.

I can provide the page https://wandb.ai/ihoholko/masks_attributs?workspace=user-ihoholko (original one) where I have the issue.

And toy example where I have the same result:
https://wandb.ai/ihoholko/MyAwesomeConfusion?workspace=user-ihoholko
And its code to allow you to reproduce it:

```python
"""
 File name   : example.py
 Description : description

 Date created : 19.04.2021
 Author:  Ihar Khakholka
"""


import os
import torch
from torch import nn
import torch.nn.functional as F
from torchvision.datasets import MNIST
from torch.utils.data import DataLoader, random_split
from torchvision import transforms
import pytorch_lightning as pl
import wandb
import numpy as np
from pathlib import Path

class LitAutoEncoder(pl.LightningModule):

    def __init__(self):
        super().__init__()
        self.encoder = nn.Sequential(nn.Linear(28 * 28, 128), nn.ReLU(), nn.Linear(128, 3))
        self.decoder = nn.Sequential(nn.Linear(3, 128), nn.ReLU(), nn.Linear(128, 28 * 28))

    def forward(self, x):
        # in lightning, forward defines the prediction/inference actions
        embedding = self.encoder(x)
        return embedding

    def training_step(self, batch, batch_idx):
        # training_step defines the train loop. It is independent of forward
        x, y = batch
        x = x.view(x.size(0), -1)
        z = self.encoder(x)
        x_hat = self.decoder(z)
        loss = F.mse_loss(x_hat, x)
        self.log('train_loss', loss)
        return loss

    def training_epoch_end(self, outputs):
        self.logger.experiment.log({'conf': wandb.plot.confusion_matrix(
            np.array([[0.5,0.3,0.2], [0.5,0.3,0.2], [0.5,0.3,0.2]]), np.array([1,1,2]),
        )})
        self.logger.experiment.log({'roc': wandb.plot.roc_curve(
            np.array([1, 1, 2]),   np.array([[0.5, 0.3, 0.2], [0.5, 0.3, 0.2], [0.5, 0.3, 0.2]]),
        )})

    def configure_optimizers(self):
        optimizer = torch.optim.Adam(self.parameters(), lr=1e-3)
        return optimizer


SAVE_DIR = Path(__file__).resolve().parents[1] / "training_logs"

if __name__ == '__main__':
    dataset = MNIST(os.getcwd(), download=True, transform=transforms.ToTensor())
    train, val = random_split(dataset, [55000, 5000])

    autoencoder = LitAutoEncoder()

    logger = pl.loggers.WandbLogger(project="MyAwesomeConfusion", save_dir=SAVE_DIR, )
    logger.watch(autoencoder)
    # logger.log_hyperparams()

    trainer = pl.Trainer(logger=logger, weights_save_path="training/logs", limit_train_batches = .1, gpus=1)
    trainer.fit(autoencoder, DataLoader(train), DataLoader(val))
```


#### requarements.txt

> pytorch-lightning==1.2.7 
torch==1.8.1
torchvision==0.9.1
wandb==0.10.25

**Comment by 2dot71mily** [comment]
Experiencing same.

**Comment by 2dot71mily** [comment]
Workspace shows a populated confusion matrix when older runs (anything before the most recent peach run), is selected:
![image](https://user-images.githubusercontent.com/21292059/115911911-8cebb400-a423-11eb-9304-cec2ce297c58.png)

However, when only peach run is selected: 
![image](https://user-images.githubusercontent.com/21292059/115912001-a987ec00-a423-11eb-9a70-d246481b0f5e.png)


workspace: https://wandb.ai/emilylearning/youtube_multi_?workspace=user-emilylearning

code: https://colab.research.google.com/drive/1olAN3mqunN6FWpgxTgL6W72zdp6Jzbcf?usp=sharing#scrollTo=0L5uBK4XhR8N
note: code unchanged between old and new (peach) runs

In more recent workspaces (where I don't have any old successful examples of conf_mat), it appears the issue is that conf_mat doesn't accept `conf_mat_table` as an acceptable key, even tho `conf_mat_table` is populated.

**Comment by KyleGoyette** [comment]
You should be able to edit your query for the confusion matrix from a summaryTable to a historyTable to have the chart show. Note that if you've logged the chart several times, you'll need to select an index in the historyTable query. For it to display properly.

We've identified the source of this issue, and are working on a fix that will prevent this from happening with future runs.

**Comment by 2dot71mily** [comment]
thanks Kyle, I'm a wandb n00b. will give a try.

**Comment by IgorHoholko** [comment]
@2dot71mily could you please share some more details what you have done? :)

**Comment by KyleGoyette** [comment]
@IgorHoholko If you click the edit modal, then modify your query to look as follows:
![image](https://user-images.githubusercontent.com/13547142/115914360-b22df180-a426-11eb-9160-84f41ab63bcd.png)
For your ROC chart in this workspace:
https://wandb.ai/ihoholko/masks_attributs?workspace=user-ihoholko 

It should work.

**Comment by IgorHoholko** [comment]
@KyleGoyette Yes, Thanks. It works in this way.
But Is it impossible for now to log it automatically? And what does the magic `index: 112` mean ?:) I don't see this number in the project.

**Comment by 2dot71mily** [comment]
Sorry, just seeing now @IgorHoholko. I think that index is an optional arg.. doesn't matter for us.

For me:
before:
![image](https://user-images.githubusercontent.com/21292059/115915865-cf63bf80-a428-11eb-8b9c-6455d284c6f4.png)


after: 
![image](https://user-images.githubusercontent.com/21292059/115915840-c7a41b00-a428-11eb-9be1-9e2924be9fc9.png)

**Comment by KyleGoyette** [comment]
@IgorHoholko The index marks the step that you logged the table at. When you do a historyTable query, we automatically query for the steps that you logged it at: 
![image](https://user-images.githubusercontent.com/13547142/115916042-1487f180-a429-11eb-8447-2505f6690c1b.png)

I just got the 112 by scrolling to the bottom of the data:
![image](https://user-images.githubusercontent.com/13547142/115916483-bad3f700-a429-11eb-842d-efdfea0f924d.png)


Filling in the index makes your chart look correct:
![image](https://user-images.githubusercontent.com/13547142/115916560-d17a4e00-a429-11eb-87e9-bd2ae11c1586.png)

**Comment by KyleGoyette** [comment]
@IgorHoholko We'll have a fix in an upcoming release to resolve this issue so you can log automatically again.

**Comment by sydholl** [comment]
This issue was fixed in April 2021. Please let us know if you're still experiencing any issues.

**Comment by MohammedAljahdali** [comment]
Hi, I am experiencing this issue atm.

This is the confusion matrix in one of the runs: 
![image](https://user-images.githubusercontent.com/24215630/157652909-8e4c814e-0308-47e2-93fd-f204cdece0c1.png)

And here it is for the other run where it does not show:
![image](https://user-images.githubusercontent.com/24215630/157653066-cc99d179-d858-4ece-8add-d6dd479be480.png)
